src/Traits/HasResetDate.php

```diff
@@ -12,14 +12,18 @@ trait HasResetDate
     /**
      * Get feature's reset date.
      *
-     * @param string $dateFrom
+     * @param Carbon|null $dateFrom
      *
      * @return \Carbon\Carbon
      * @throws \Exception
      */
-    public function getResetDate(Carbon $dateFrom): Carbon
+    public function getResetDate(?Carbon $dateFrom = null): Carbon
     {
-        $period = new Period($this->resettable_interval, $this->resettable_period, $dateFrom ?? now());
+        $today = Carbon::now();
+
+        do {
+            $period = new Period($this->resettable_interval, $this->resettable_period, $dateFrom ?? $today);
```

Comment:
Don't we miss here assigning a new value for $dateFrom? The new $dateFrom should be equal to the end of the previous period?

Reply:
Oh mama

Reply:
Solved v4 and main, thanks for noticing, sometimes at the end of the day one does not know what he does